src/symbol/placement.js

```diff
@@ -773,38 +779,35 @@ export class Placement {
 
             const placedOrientation = this.placedOrientations[symbolInstance.crossTileID];
-            const verticalHidden = (placedOrientation === WritingMode.horizontal || placedOrientation === WritingMode.horizontalOnly) ? 1 : 0;
-            const horizontalHidden = placedOrientation === WritingMode.vertical ? 1 : 0;
+            const horizontalHidden = placedOrientation === WritingMode.vertical;
+            const verticalHidden = placedOrientation === WritingMode.horizontal || placedOrientation === WritingMode.horizontalOnly;
 
             if (hasText) {
                 const packedOpacity = packOpacity(opacityState.text);
                 // Vertical text fades in/out on collision the same way as corresponding
                 // horizontal text. Switch between vertical/horizontal should be instantaneous
                 const horizontalOpacity = horizontalHidden ? PACKED_HIDDEN_OPACITY : packedOpacity;
-                for (let i = 0; i < numHorizontalGlyphVertices / 4; i++) {
-                    bucket.text.opacityVertexArray.emplaceBack(horizontalOpacity);
-                }
+                addOpacities(bucket.text, numHorizontalGlyphVertices, horizontalOpacity);
                 const verticalOpacity = verticalHidden ? PACKED_HIDDEN_OPACITY : packedOpacity;
-                for (let i = 0; i < numVerticalGlyphVertices / 4; i++) {
-                    bucket.text.opacityVertexArray.emplaceBack(verticalOpacity);
-                }
+                addOpacities(bucket.text, numVerticalGlyphVertices, verticalOpacity);
+
                 // If this label is completely faded, mark it so that we don't have to calculate
                 // its position at render time. If this layer has variable placement, shift the various
                 // symbol instances appropriately so that symbols from buckets that have yet to be placed
                 // offset appropriately.
-                const symbolHidden = opacityState.text.isHidden() ? 1 : 0;
+                const symbolHidden = opacityState.text.isHidden();
                 [
                     symbolInstance.rightJustifiedTextSymbolIndex,
                     symbolInstance.centerJustifiedTextSymbolIndex,
                     symbolInstance.leftJustifiedTextSymbolIndex
                 ].forEach(index => {
                     if (index >= 0) {
-                        bucket.text.placedSymbolArray.get(index).hidden = symbolHidden || horizontalHidden;
+                        bucket.text.placedSymbolArray.get(index).hidden = Number(symbolHidden || horizontalHidden);
                     }
                 });
 
                 if (symbolInstance.verticalPlacedTextSymbolIndex >= 0) {
-                    bucket.text.placedSymbolArray.get(symbolInstance.verticalPlacedTextSymbolIndex).hidden = symbolHidden || verticalHidden;
+                    bucket.text.placedSymbolArray.get(symbolInstance.verticalPlacedTextSymbolIndex).hidden = Number(symbolHidden || verticalHidden);
```

Comment:
`Number` is apparently very slow.
https://jsperf.com/boolean-int-conversion/3
`| 0` seems better and more concise.

Reply:
I'm not sure the performance difference is important here but I switched to a ternary expression since that seems clearer.

Reply:
yea, I don't think it really would make a difference.